test/behavior/for.zig

```diff
@@ -535,3 +535,11 @@ test "return from inline for" {
     };
     try std.testing.expect(!S.do());
 }
+
+test "for loop 0 length range" {
+    const map: []const u8 = &.{};
+    for (map, 0..map.len) |i, j| {
+        _ = i;
+        _ = j;
+    }
+}
```

Comment:
This needs to actually test something, such as that the loop body never executes.